tmva/sofie/inc/TMVA/ROperator_Tile.hxx

```diff
@@ -0,0 +1,108 @@
+#ifndef TMVA_SOFIE_ROPERATOR_TILE
+#define TMVA_SOFIE_ROPERATOR_TILE
+
+#include "TMVA/SOFIE_common.hxx"
+#include "TMVA/ROperator.hxx"
+#include "TMVA/RModel.hxx"
+
+#include <cassert>
+#include <sstream>
+
+namespace TMVA {
+namespace Experimental {
+namespace SOFIE {
+
+template <typename T>
+class ROperator_Tile final : public ROperator {
+
+private:
+    std::string fNData;        // input data tensor name
+    std::string fNReps;        // repetitions tensor name
+    std::string fNOutput;      // output tensor name
+    std::vector<size_t> fShapeInput; // input shape data
+    std::vector<size_t> fShapeOutput; // output shape data
+
+public:
+    ROperator_Tile() {}
+    ROperator_Tile(std::string nameData, std::string nameReps, std::string nameOutput)
+        : fNData(UTILITY::Clean_name(nameData)), fNReps(UTILITY::Clean_name(nameReps)),
+          fNOutput(UTILITY::Clean_name(nameOutput)) {}
+
+    // output type is same as input
+    std::vector<ETensorType> TypeInference(std::vector<ETensorType> input) {
```

Comment:
`input` should be passed by const reference as it's potentially expensive to copy